frontend/src/components/test-history.js

```diff
@@ -33,334 +31,317 @@ import {
+      updateFilters(id, null, null);
+    }
   };
 
-  getHistorySummary () {
-    // get the passed/failed/etc test summary
-    let filters = {...this.state.filters};
-    // disregard result filter (we want all results)
-    delete filters['result'];
-    let api_filter = toAPIFilter(filters).join();
-    let dataToSummary = Object.assign({
-      'passed': 'passes',
-      'failed': 'failures',
-      'error': 'errors',
-      'skipped': 'skips',
-      'xfailed': 'xfailures',
-      'xpassed': 'xpasses'
-    });
-    let summary = Object.assign({
-      'passes': 0,
-      'failures': 0,
-      'errors': 0,
-      'skips': 0,
-      'xfailures': 0,
-      'xpasses': 0
-    });
-
-    HttpClient.get(
-      [Settings.serverUrl, 'widget', 'result-aggregator'],
-      {
-        group_field: 'result',
-        additional_filters: api_filter,
-      }
-    )
-      .then(response => HttpClient.handleResponse(response))
-      .then(data => {
-        data.forEach(item => {
-          summary[dataToSummary[item['_id']]] = item['count'];
-        });
-        this.setState({historySummary: summary}, this.getResultsForTable);
-      });
-  }
 
-  getLastPassed (){
+  useEffect(() => {
     // get the passed/failed/etc test summary
-    let filters = {...this.state.filters};
-    // disregard result filter so we can filter on last passed
-    delete filters['result'];
-    delete filters['start_time'];
-    filters['result'] = {'op': 'eq', 'val': 'passed'};
-    let params = buildParams(filters);
-    params['filter'] = toAPIFilter(filters);
-    params['pageSize'] = 1;
-    params['page'] = 1;
-    params['estimate'] = 'true';
-
-    HttpClient.get([Settings.serverUrl, 'result'], params)
-      .then(response => HttpClient.handleResponse(response))
-      .then(data => this.setState({
-        lastPassedDate:
-            <React.Fragment>
-              <Link target="_blank" rel="noopener noreferrer" to={`../results/${data.results[0].id}`} relative="Path">
-                <Badge isRead>
-                  {new Date(data.results[0].start_time).toLocaleString()}
-                </Badge>
-              </Link>
-            </React.Fragment>
-      }))
-      .catch((error) => {
-        console.error('Error fetching result data:', error);
-        this.setState({lastPassedDate: <Badge isRead>n/a</Badge>});
-      });
-  }
+    if (JSON.stringify(filtersState) !== '{}') {
+      let historyFilters = {...filtersState};
+      // disregard result filter (we want all results)
+      delete historyFilters['result'];
+      let api_filter = toAPIFilter(historyFilters).join();
 
-  getResultsForTable () {
-    if (this.state.comparisonResults !== undefined) {
-      this.setState({
-        results: this.state.comparisonResults,
-        rows: this.state.comparisonResults.map((result, index) => resultToTestHistoryRow(result, index, this.setFilter)).flat(),
+      let summary = Object.assign({
```

Comment:
Nitpick: 

1. Could be const since we are not re-assigning object but changing it's props
2. No need to use Object.assign